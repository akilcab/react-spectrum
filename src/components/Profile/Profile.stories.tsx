import { Story, Meta } from "@storybook/react/types-6-0";
import {Profile, ProfileProps} from "./Profile";

export default {
  title: "Profile",
  component: Profile,
  description: `A Profile.`,
  argTypes: {
    logo: { control: "text" },
    username: { control: "text" },
    email: { control: "text" },
  },
} as Meta;

//👇 We create a “template” of how args map to rendering
const Template: Story<ProfileProps> = (
  args
) => <Profile {...args}/>;

//👇 Each story then reuses that template
export const Default = Template.bind({});
Default.args = {
  username: "Amara Bamba",
  email: "amara.bamba@akiltechnologies.com",
};
