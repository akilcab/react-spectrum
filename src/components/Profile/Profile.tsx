import React from "react";
import {
  Avatar,
  Badge,
  ListItem,
  ListItemAvatar,
  ListItemText,
} from "@material-ui/core";

export type ProfileProps = {
  username: string;
  email: string;
  logo: string;
};

export const Profile: React.FC<ProfileProps> = (props) => {
  const { username, email, logo } = props;

  return (
    <ListItem>
      <ListItemAvatar>
        <Badge variant="dot" color="primary">
          <Avatar src={logo} variant="rounded" />
        </Badge>
      </ListItemAvatar>
      <ListItemText primary={username} secondary={email} />
    </ListItem>
  );
};
