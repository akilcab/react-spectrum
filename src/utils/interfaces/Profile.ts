export interface Profile {
  avatar: string;
  createdAt: Date;
  createdBy: string;
  id: string;
  isBlocked: boolean;
  isEnabled: boolean;
  isFirstUser: string;
  lastConnexion: Date;
  updatedAt: Date;
  updatedBy: string;
}
