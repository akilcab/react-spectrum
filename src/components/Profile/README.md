# Description of component

Profile is a component to display the logged-in user on the dashboard.

#### Example :

![](assets/desc.png)

## How to import

```node
import { Profile } from "app/shared/components/Profile";
```

## How to use

```node
....
<Profile
  username={"Amara Bamba"}
  logo={"./avatar.png"}
  email={"amara.dianney@gmail@gmail.com"}
/>
....
```

## Props

| Props       | Required | Description | Type |
| :---        | :---     | :---        |:---  |
| `username`       | true    | Taking a user full name | `string` |
| `logo`       | true    | Taking user logo file | `string` |
| `email`       | true    | Taking a valid email| `string` |

## CSS properties we can override

You can override all CSS properties of the ListProps type of Material UI.
