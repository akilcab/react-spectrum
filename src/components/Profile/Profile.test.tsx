import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Profile } from "./Profile";

describe("<Profile />", () => {
  test("rend correctement", () => {
    const component = renderer
      .create(
        <Profile
          username={"Amara Bamba"}
          logo={""}
          email={"amara.dianney@gmail@gmail.com"}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  test("should have content texts in the component", () => {
    const component = render(
      <Profile
        username={"Amara Bamba"}
        logo={""}
        email={"amara.dianney@gmail@gmail.com"}
      />
    );
    expect(component.container).not.toHaveTextContent("Bamba Amara");
    expect(component.container).toHaveTextContent("Amara Bamba");
    expect(component.container).not.toHaveTextContent(
      "dianney.amara@gmail@gmail.com"
    );
    expect(component.container).toHaveTextContent(
      "amara.dianney@gmail@gmail.com"
    );
  });

  test("try to display the profile without its require props", () => {
    expect(Profile).toThrowError();
  });
});
