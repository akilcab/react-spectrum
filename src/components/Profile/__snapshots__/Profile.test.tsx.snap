// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> rend correctement 1`] = `
<li
  className="MuiListItem-root MuiListItem-gutters"
  disabled={false}
>
  <div
    className="MuiListItemAvatar-root"
  >
    <span
      className="MuiBadge-root"
    >
      <div
        className="MuiAvatar-root MuiAvatar-rounded MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <span
        className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorPrimary MuiBadge-dot"
      >
        
      </span>
    </span>
  </div>
  <div
    className="MuiListItemText-root MuiListItemText-multiline"
  >
    <span
      className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
    >
      Amara Bamba
    </span>
    <p
      className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
    >
      amara.dianney@gmail@gmail.com
    </p>
  </div>
</li>
`;
